File: Sentimental-readability/readability.py
from cs50 import get_string


def main():
    text = get_string("Text: ")

    letter_count, word_count, sentence_count = 0, 0, 0

    length = len(text)
    for i in range(length):
        if text[i].isalpha():
            letter_count += 1

        if text[i] == ' ':
            word_count += 1

        if text[i] in ('.', '!', '?'):
            sentence_count += 1

    # Count the last word
    word_count += 1

    L = (letter_count / word_count) * 100
    S = (sentence_count / word_count) * 100

    index = 0.0588 * L - 0.296 * S - 15.8

    grade_level = round(index)

    if grade_level < 1:
        print("Before Grade 1")
    elif grade_level > 16:
        print("Grade 16+")
    else:
        print(f"Grade {grade_level}")


if __name__ == "__main__":
    main()

File: readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(void)
{
    string text = get_string("Text: ");

    int letter_count = 0;
    int word_count = 0;
    int sentence_count = 0;

    int length = strlen(text);
    for (int i = 0; i < length; i++)
    {

        if (isalpha(text[i]))

        {
            letter_count++;
        }

        if (text[i] == ' ')

        {
            word_count++;
        }

        if (text[i] == '.' || text[i] == '!' || text[i] == '?')

        {
            sentence_count++;
        }
    }
    word_count += 1;

    float L = ((float) letter_count / word_count) * 100;
    float S = ((float) sentence_count / word_count) * 100;

    float index = 0.0588 * L - 0.296 * S - 15.8;

    int grade_level = round(index);

    if (grade_level < 1)
    {
        printf("Before Grade 1\n");
    }

    else if (grade_level > 16)
    {
        printf("Grade 16+\n");
    }

    else
    {
        printf("Grade %i\n", grade_level);
    }
}
